### PR TITLE
Add migration guide

### PR DESCRIPTION
**Pull request by dmlap**
Create some instructions for plugins moving from 1.0 to 2.0. Update the changelog.

### CODE REVIEW

**Comment by gkatsev** [comment]
LGTM

**Comment by apadhye** [comment]
:+1:

File: migrating-to-2.0.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Guidance for Ad Partners
+One of the best features of video.js is the community of plugins and customizations that has built up around it.
+Ad support is an important part of that ecosystem but not all plugin authors write with advertisements in mind (and we probably won't be able to convince them to start).
+Rather than throw our hands in the air, we're making some big changes in [videojs-contrib-ads](https://github.com/videojs/videojs-contrib-ads) to make advertisements more compatible with the rest of the video.js universe.
+In version 2.0 of the plugin, we redispatch events with different prefixes depending on whether an ad is playing or not.
+When an ad is playing, events are prefixed with `ad` and when content is resuming after an ad break, events are prefixed with `content`.
```
Comment (bclwhitaker) [inline migrating-to-2.0.md]: Might be helpful to provide an example or two on how this prefix looks on an event in both cases here. Sometimes prefixes are hyphenated, etc.  You provide examples in the migration guide, but not everyone is migrating...

**Comment by heff** [comment]
This is great.

**Pull request merged**